embassy-stm32/src/rcc/mod.rs

```diff
@@ -196,11 +202,15 @@ impl RccInfo {
     pub(crate) fn disable_with_cs(&self, _cs: CriticalSection) {
         if self.refcount_idx_or_0xff != 0xff {
             let refcount_idx = self.refcount_idx_or_0xff as usize;
-            unsafe {
-                crate::_generated::REFCOUNTS[refcount_idx] -= 1;
-            }
-            if unsafe { crate::_generated::REFCOUNTS[refcount_idx] } > 0 {
-                return;
+
+            // Use .get_mut instead of []-operator so that we control how bounds checks happen.
+            // Otherwise, core::fmt will be pulled in here in order to format the integer in the
+            // out-of-bounds error.
+            if let Some(refcount) = unsafe { crate::_generated::REFCOUNTS }.get_mut(refcount_idx) {
```

Comment:
this is missing the `else`, we don't want to silently ignore wrong indexes.

Reply:
Hm true, although this didn't show up in my profiling which is why I forgot to add it.  I guess the compiler can prove that the index is valid since `enable_and_reset_with_cs` succeeded, maybe? Or maybe it gets confused since mutations are happening outside `UnsafeCell`, which might be UB in this situation?  I'll restore the panic for sure though!